hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFile.java

```diff
@@ -614,7 +629,11 @@ public static Reader createReader(FileSystem fs, Path path, CacheConfig cacheCon
   static Reader createReaderFromStream(Path path, FSDataInputStream fsdis, long size,
       CacheConfig cacheConf, Configuration conf) throws IOException {
     FSDataInputStreamWrapper wrapper = new FSDataInputStreamWrapper(fsdis);
-    return openReader(path, wrapper, size, cacheConf, null, true, conf);
+    Reader preadReader = openReader(path, wrapper, size, cacheConf, null, true, true, conf);
```

Comment:
This is a bit strange, that we create two readers, but only returns one of them, without closing the other one...I know there is no problem, as the resources are in the FSDataInputStream passed in, but still a bit strange

Reply:
This method is for UT only, may be we can move it to TestHFile.